mobly/controllers/android_device.py

```diff
@@ -648,13 +649,14 @@ def _terminate_sl4a(self):
         Send terminate signal to sl4a server; stop dispatcher associated with
         the session. Clear corresponding droids and dispatchers from cache.
         """
+        if self.ed:
```

Comment:
Like David said, it makes it awkward to tear down sl4a, then tear down ed, then go back and stop the app. Also it's surprising to kill an sl4a session and then kill something that's using it. But I will revert the logic pending a broader refactor in #33.